src/com/google/javascript/jscomp/PeepholeFoldConstants.java

```diff
@@ -1151,16 +1174,21 @@ private static TernaryValue compareAsNumbers(int op, Node left, Node right, bool
     double rv = rightValue;
 
     switch (op) {
+      // See http://www.ecma-international.org/ecma-262/6.0/#sec-strict-equality-comparison
       case Token.SHEQ:
-      case Token.EQ:
         Preconditions.checkState(
             left.isNumber() && right.isNumber());
         return TernaryValue.forBoolean(lv == rv);
       case Token.SHNE:
-      case Token.NE:
         Preconditions.checkState(
             left.isNumber() && right.isNumber());
         return TernaryValue.forBoolean(lv != rv);
+      // See http://www.ecma-international.org/ecma-262/6.0/#sec-abstract-equality-comparison
+      case Token.EQ:
```

Comment:
I'm a little bit wary of making this method more permissive. Before, if you called it on `0 == null`, you would get a Preconditions failure, but now you'll get the (incorrect) value of true.